tools/replay/desktop_main.cpp

```diff
@@ -88,29 +89,82 @@ void WaitForExit() {}
 
 const char kLayerEnvVar[] = "VK_INSTANCE_LAYERS";
 
-#if defined(D3D12_SUPPORT)
-bool BrowseFile(const std::string&                           input_filename,
-                const gfxrecon::decode::DumpResourcesTarget& dump_resources_target,
-                gfxrecon::decode::TrackDumpDrawCall&         out_track_dump_target)
+void RunVulkanPreProcessConsumer(const std::string&                      input_filename,
+                                 gfxrecon::decode::VulkanReplayOptions&  replay_options,
+                                 gfxrecon::decode::VulkanReplayConsumer& replay_consumer)
 {
-    gfxrecon::decode::TrackDumpDrawCall* track_dump_target = nullptr;
+    gfxrecon::decode::FileProcessor file_processor;
+    if (file_processor.Initialize(input_filename))
+    {
+        gfxrecon::decode::VulkanPreProcessConsumer pre_process_consumer;
+
+        bool process_dump_resources =
+            replay_options.enable_dump_resources && replay_options.dump_resources_block_indices.empty();
+        if (process_dump_resources)
+        {
+            pre_process_consumer.EnableDumpResources(replay_options.dump_resources_target);
+        }
+
+        gfxrecon::decode::VulkanDecoder decoder;
+        decoder.AddConsumer(&pre_process_consumer);
+        file_processor.AddDecoder(&decoder);
+        file_processor.ProcessAllFrames();
```

Comment:
Does this mean that the whole capture file will be read in in order to generated the dump parameters? Would it be possible if the dump occurs early in the capture file to end this preprocessing step  prematurely to avoid reading the whole file?

Reply:
`VulkanPreProcessConsumer pre_process_consumer` became necessary to run because it isn't only for dump, also for checking if this replay is Vulkan or dx12.  See `replay_options.enable_vulkan = pre_process_consumer.WasVulkanAPIDetected();`. 

Plus, It checks dump resources early
```
if (process_dump_resources)
{
    pre_process_consumer.EnableDumpResources(replay_options.dump_resources_target);
}
```
If it's not enabled, `IsComplete` will return true after knowing its Vulkan or dx12 replay, and quit this process.
```
virtual bool IsComplete(uint64_t block_index) override
{
    return check_vulkan_consumer_usage_complete_ && (check_dump_resources_complete_ || !enable_dump_resources_);
}
```